hpack/table.py

```diff
@@ -172,12 +185,15 @@ def search(self, name, value):
         """
         offset = HeaderTable.STATIC_TABLE_LENGTH + 1
         partial = None
-        for (i, (n, v)) in enumerate(HeaderTable.STATIC_TABLE):
-            if n == name:
-                if v == value:
-                    return i + 1, n, v
-                elif partial is None:
-                    partial = (i + 1, n, None)
+
+        header_name_search_result = HeaderTable.STATIC_TABLE_MAPPING.get(name)
+        if header_name_search_result is not None:
+            index = header_name_search_result.positions.get(value)
+            if index is not None:
+                return index + 1, name, value
+            else:
+                partial = (header_name_search_result.min + 1, name, None)
```

Comment:
Would it not be the same amount of dicts that's being used in this PR minus the `_HeaderNameSearchResult` objects? There's one dict with names and a dict per name with values contained in the `_HeaderNameSearchResult` instance.

Perhaps we could minimize to one dict where you attempt `d[(name, value)]` first and if that fails then you attempt `d[(name, None)]`?